src/python/pants/task/testrunner_task_mixin.py

```diff
@@ -63,6 +66,88 @@ def execute(self):
       all_targets = self._get_targets()
       self._execute(all_targets)
 
+  def report_test_info(self, scope, target, test_name, test_info):
+    """Add test information to target information
+
+    :param string scope: The running scope
+    :param Target target: The target that we want to store the test info under
+    :param string test_name: The key (test name) for the info being stored
+    :param dict test_info: The info being stored
+    """
+    if target and scope:
+      address = target.address.spec
+      target_type = target.type_alias
+      self.context.run_tracker.report_target_info('GLOBAL', address, ['target_type'], target_type)
+      self.context.run_tracker.report_target_info(scope, address, [test_name], test_info)
+
+  @staticmethod
+  def parse_test_info(xml_path, error_handler, additional_testcase_attributes=None):
```

Comment:
I believe the two you are thinking of parse failed targets for each failed test and not the actual test information. See [here](https://github.com/pantsbuild/pants/blob/ae1ea88d5a280120dff3a4cff2141b774ccca139/src/python/pants/backend/python/tasks2/pytest_run.py#L438) and [here](https://github.com/pantsbuild/pants/blob/ae1ea88d5a280120dff3a4cff2141b774ccca139/src/python/pants/java/junit/junit_xml_parser.py#L148).